src/training/vqa_fine_tune.py

```diff
@@ -0,0 +1,311 @@
+        text = item["question"]
+        label = self.labels[idx]
+        return {
+            'image': self.transforms(image),
+            'text': self.tokenize([text])[0],
+            'label': torch.tensor(label)
+        }
+
+def get_task_dataloaders(df, transforms, args):
+    tokenizer = get_tokenizer(args.model)
+
+    dataloaders = {}
+    dataset = VQATextDataset(df,
+        "train",
+        transforms,
+        tokenizer=tokenizer,
+    )
+    dataloader = DataLoader(
+        dataset,
+        batch_size=args.batch_size,
+        shuffle=True,
+        num_workers=args.workers,
+        pin_memory=True,
+        drop_last=True,
+    )
+    dataloaders["train"] = dataloader
+
+    return dataloaders
+
+class CLIPMultimodalClassifier(nn.Module):
+    def __init__(self, encoder, embed_dim, num_labels):
+        super().__init__()
+
+        self.encoder = encoder
+        
+        self.fc1 = nn.Linear(embed_dim * 2, 1536) #size of answer space
+        self.lnorm = nn.LayerNorm(1536)
+        self.fc2 = nn.Linear(1536, num_classes)
+    def forward(self, image, text):
+        # CLIP doesn't have a multimodal encoder, so we concatenate the features
+        text_features = self.encoder.encode_text(text)
+        image_features = self.encoder.encode_image(image)
+        multimodal_features = torch.cat([image_features, text_features], dim=-1)
+        layer = self.lnorm(F.relu(self.fc1(multimodal_features)))
+        logits = self.fc2(layer)
+        return logits
+
+class EarlyStopping:
+
+    def __init__(self, patience=5, threshold=0.0, metric_name="accuracy"):
+        self.patience = patience
+        self.threshold = threshold
+        self.patience_counter = 0
+        self.best_score = None
+        self.best_metrics = None
+        self.metric_name = metric_name
+
+    def step(self, metrics):
+        score = metrics[self.metric_name]
+        if self.best_score is None:
+            self.best_score = score
+            self.best_metrics = metrics
+        elif score < self.best_score + self.threshold:
+            self.patience_counter += 1
+        else:
+            self.best_score = score
+            self.best_metrics = metrics
+            self.patience_counter = 0
+
+        if self.patience_counter >= self.patience:
+            return True
+        return False
+
+def compute_metrics(model, dataloader, device, args):
+    model.eval()
+    metric = evaluate.load("accuracy")
+    val_loss = 0
+    samples_seen = 0
+    loss_fn = nn.CrossEntropyLoss()
+    for batch in dataloader:
+        with torch.no_grad():
+            image = batch["image"].to(device)
+            text = batch["text"].to(device)
+            label = batch["label"].to(device)
+            samples_seen += text.shape[0]
+            logits = model(image, text)
+
+            #predictions = torch.argmax(logits)
+            batch_val_loss = loss_fn(logits, label)
+        val_loss += batch_val_loss.item()
+        #metric.add_batch(
```

Comment:
are you using accuracy as the metric? it doesn't seem that you're computing it here.

Reply:
how is the `test-dev VQA score` (as mentioned in the FLAVA paper) computed?